core/injected-scripts/domReplayer.ts

```diff
@@ -0,0 +1,410 @@
+// NOTE: do not use node dependencies
+
+import type { IFrontendDomChangeEvent } from '@secret-agent/core/models/DomChangesTable';
+
+declare global {
+  interface Window {
+    replayDomChanges(...args: any[]);
+    replayInteractions(...args: any[]);
+    getIsMainFrame?: () => boolean;
+    idMap: Map<number, Node>;
+    debugLogs: any[];
+    debugToConsole: boolean;
+    getNodeById(id: number): Node;
+  }
+}
+
+// copied since we can't import data types
```

Comment:
Why can't data types be imported?

Reply:
This is an injected script - gets run in the browser...

Reply:
So it's converted to javascript, right? And typescript's enums are converted to a JS object.